src/lib/maps/ColorLegend/ColorLegend.svelte

```diff
@@ -0,0 +1,381 @@
+        <g class="legend-inner" transform="translate({margin.left}, {margin.top})">
+          {#if scaleType === "continuous"}
+            <defs>
+              <linearGradient id={scaleId}>
+                {#each range as color, i}
+                  <stop offset="{(100 / (range.length - 1)) * i}%" stop-color={color} />
+                {/each}
+              </linearGradient>
+            </defs>
+            <rect
+              fill="url(#{scaleId})"
+              width={legendWidth}
+              {height}
+              y={tickPosition === "bottom" ? 0 : tickSize + tickMargin}
+            ></rect>
+          {:else if scaleType === "sequential"}
+            <defs>
+              <linearGradient id={scaleId}>
+                {#each domain as stop, i}
+                  <stop offset="{100 * (i / (domain.length - 1))}%" stop-color={scale(stop)} />
+                {/each}
+              </linearGradient>
+            </defs>
+            <rect
+              fill="url(#{scaleId})"
+              width={legendWidth}
+              {height}
+              y={tickPosition === "bottom" ? 0 : tickSize + tickMargin}
+            ></rect>
+          {:else if scaleType === "threshold"}
+            {#each range as color, i}
+              <rect x={xScale(i - 1)} y="0" width={xScale(i) - xScale(i - 1)} {height} fill={color}
+              ></rect>
+            {/each}
+          {:else if scaleType === "ordinal"}
+            {#each domain as cat, i}
+              <rect
+                x={xScale(cat)}
+                y={tickPosition === "bottom" ? 0 : tickSize + tickMargin}
+                width={Math.max(0, xScale.bandwidth() - 1)}
+                {height}
+                fill={scale(cat)}
+              ></rect>
+            {/each}
+          {/if}
+          {#if scaleType && width && xScale}
+            <g class="legend-ticks">
+              {#each legendTicks as tick}
+                {@const xPosition =
+                  scaleType === "ordinal" ? xScale(tick) + xScale.bandwidth() / 2 : xScale(tick)}
+                {@const yPosition = tickPosition === "top" ? 0 : height + tickMargin + tickSize}
+                {#if scaleType !== "ordinal" && tickLineWidth}
+                  <line
+                    x1={xPosition}
+                    x2={xPosition}
+                    y1={tickPosition === "bottom" ? 0 : height + tickMargin + tickSize}
+                    y2={tickPosition === "bottom" ? height + tickMargin : tickSize}
+                    stroke={tickLineColor}
+                    stroke-width={tickLineWidth}
+                  ></line>
+                {/if}
+                <text
+                  fill="black"
+                  alignment-baseline={tickPosition === "top" ? "hanging" : "baseline"}
+                  y={yPosition}
+                  x={xPosition}
+                  font-size="{tickSize}px"
+                  text-anchor="middle">{tickFormatFn(tick)}</text
```

Comment:
consider adding a prop/some sort of config for modifying `text-anchor`'s value. "end" for last value and "start" for first, suggestion below would require adding an index variable `i` in `{#each}`
```suggestion
text-anchor={i === legendTicks.length ? "end" : i === 0 ? "start" : "middle"}>{tickFormatFn(tick)}</text
```

Reply:
This is a good call. I might need to try something a bit more robust than just the first/last tick defaulting to "end" and "start". There are some cases where that wouldn't be desirable (cases where the first tick doesn't align with the min of the scale and where the last tick doesn't align with the max of the scale) but we should definitely expose a bit more control as well as some better defaults.